dbms/src/Storages/Transaction/Collator.cpp

```diff
@@ -402,7 +443,15 @@ class UnicodeCICollator final : public ITiDBCollator
 
     StringRef sortKey(const char * s, size_t length, std::string & container) const override
     {
-        auto v = rtrim(s, length);
+        std::string_view v;
+        if (padding)
+        {
+            v = rtrim(s, length);
+        }
+        else
+        {
+            v = std::string_view(s, length);
+        }
```

Comment:
Better to have use a function todo this, so the we don't need to copy code here

Reply:
Done. PTAL @windtalker